### PR TITLE
Update.md: Add step to upgrade to latest release before running DB Assembler tool

### PR DESCRIPTION
**Pull request by steenburgh**
Add a step to the 'How to update AzerothCore to the latest stable version' (Update.md) page to upgrade to latest release before running the DB Assembler tool

## Motivation and Context
It's not immediately apparent from the steps on the 'Updating to the latest version' page that these steps will not work properly between releases. For example, recently I had to upgrade from a version from early January 2021, and I was struggling trying to figure out why it wasn't working - the DB Assembler was running, but my DB was not being upgraded. After ~4 hours of troubleshooting, I figured out that the issue was that version 3.0.0 was released in late January, and I just had to upgrade to 3.0.0 before applying the latest updates.

I figured I'd update the documentation so that nobody else runs into a similar problem in the future

Here's a link to the discord thread for the full context... consisting mostly of my own 4am ramblings but maybe also a bit of useful discussion too: https://discord.com/channels/217589275766685707/369084351129845760/851532824888606770

### CODE REVIEW

File: docs/Update.md [hunk 1]
```diff
@@ -1,27 +1,32 @@
 # How to update AzerothCore to the latest stable version
 
-## 1) Update your local sources
+## 1) Verify that you're on the latest release
+Check the [releases](https://github.com/azerothcore/azerothcore-wotlk/releases) page on the azerothcore-wotlk github page to see if there has been a new release since you last upgraded. If there has been a new release, follow the upgrade instructions to update to the latest release. **The next steps will not work properly unless you are on the latest release.**
```
Comment (Kitzunu) [inline docs/Update.md:4]: Nobody should look at releases as people will just download it. it is bad.

just link to this instead https://github.com/azerothcore/wiki/blob/master/docs/Upgrade-from-pre-3.0.0-to-latest-master.md

Comment (steenburgh) [reply docs/Update.md:4]: I'll put that in, but what's our plan for the next time we release? I'm concerned that when we release 4.0.0, we'll forget to update this 'Update.md' page with a link to the 'Upgrade from pre-4.0.0 to latest master'

File: docs/Update.md [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
+## 4) Update the Database
 
 **If you are updating your production environment, BACKUP your databases BEFORE updating them!**
 
 You need to import the **new** sql update files (located at https://github.com/azerothcore/azerothcore-wotlk/tree/master/data/sql/updates) to each database.
 
 To automate this process you can use the db-assembler bash script `bash apps/db_assembler/db_assembler.sh`.
 
-If this is the first time you are using it [READ THIS FIRST](https://www.azerothcore.org/wiki/Database-Setup) and properly configure your `conf/config.sh` file.
+If this is the first time you are using it [READ THIS FIRST](https://www.azerothcore.org/wiki/database-installation) and properly configure your `conf/config.sh` file.
```
Comment (Kitzunu) [inline docs/Update.md:30]: only do (database-installation)

File: docs/Update.md (hunk 1 above)
Comment (Kitzunu) [reply docs/Update.md:4]: Shin dont forget that, also that will be in a year or so

File: docs/Update.md [hunk 3]
```diff
@@ -1,27 +1,32 @@
 # How to update AzerothCore to the latest stable version
 
-## 1) Update your local sources
+## 1) Verify that you're on the latest release
+The latest release is currently 3.0.0. If you haven't upgraded to version 3.0.0, follow [this guide](Upgrade-from-pre-3.0.0-to-latest-master.md) to update. **The next steps will not work properly unless you are on the latest release.**
```
Comment (Kitzunu) [inline docs/Update.md:4]:
```suggestion
1. Type `.server debug` in your console
  1. If it outputs `Using World DB: ACDB 335.4-dev` then you can proceed to [Update your local source](#update-your-local-source)
  1. If it outputs anything else for `Using World DB:` then you first need to follow [this guide on updating to latest master](upgrade-from-pre-3.0.0-to-latest-master.md)
```

**Review by Kitzunu — APPROVED** [review]

**Comment by Kitzunu** [comment]
I will merge when other PR have been merged to not break this

**Comment by steenburgh** [comment]
Thanks for making those extra changes - didn't know about `.server debug`!

File: docs/Update.md [hunk 4]
```diff
@@ -1,27 +1,37 @@
 # How to update AzerothCore to the latest stable version
 
-## 1) Update your local sources
+## Verify that you're on the latest release
+
+1. Type `.server debug` in your console.
+  
+    1. If it outputs `Using World DB: ACDB 335.4-dev` then you can proceed to [Update your local source](#update-your-local-source).
```
Comment (steenburgh) [inline docs/Update.md:7]: Maybe this PR isn't the right place for this, but I've run all the upgrade scripts and my version still says 335.3, The only spot I see a SQL statement to set the version is the initial insert statement in data/sql/base/db_world/version.sql, which I don't think is run during upgrades

Comment (Kitzunu) [reply docs/Update.md:7]: I usually drop my entire db when I update so I can't say :shrug:

Comment (steenburgh) [reply docs/Update.md:7]: How do you keep your player's data intact?

Comment (Kitzunu) [reply docs/Update.md:7]: I only use it for Dev, but worldserver is disposable so I only drop that as it's rhe only one I usually edit to fix stuff. And then let the assambler update the char/auth dbs the few times they are updated.

Comment (steenburgh) [reply docs/Update.md:7]: Does `world` just not store any state? Like I could drop that and recreate it whenever I want, even as say... An owner of a 100+ player server?

Comment (Kitzunu) [reply docs/Update.md:7]: World is designed to be disposable, only auth and char dbs hold player data

**Pull request merged**